src/popup/router/index.ts

```diff
@@ -76,6 +75,10 @@ router.beforeEach(async (to, from, next) => {
     return;
   }
 
+  if (isSecureLoginEnabled.value && !isAuthenticated.value) {
+    await openAuthModal();
+  }
```

Comment:
As this test is repeated few times in the PR consider creating some kind of method that would do the check and eventually open the modal: `checkUserAuth()`